docs/vectors/c-source/sealbox_test_vectors.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017 Donald Stufft and individual contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Test vector generator/checker for libsodium's box_seal APIs
+ * to build in a unix-like environment, use a command line like
+ * $ cc sealbox_test_vectors.c -I${IPATH} -L${LPATH} -lsodium -o sealbox_test_vectors
+ * with IPATH and LPATH defined to respectively point to libsodium's include path
+ * and to the directory containing the link library libsodium.a or libsodium.o
+ *
+ */
+#include <stdio.h>
+#include <string.h>
+#include <sodium.h>
+
+int checkone (char *hxsecret, char *hxpub, char *hxplaintext, char *hxencrypted) {
+
+    int pklen = crypto_box_PUBLICKEYBYTES;
+    int sklen = crypto_box_SECRETKEYBYTES;
+    int txtlen = strlen (hxplaintext) / 2;
+    int encrlen = strlen (hxencrypted) / 2;
+
+    char *skr = sodium_malloc (sklen);
+    char *pub = sodium_malloc (pklen);
+    char *txt = sodium_malloc (txtlen);
+    char *crpt = sodium_malloc (encrlen);
```

Comment:
Could these variable names be changed to more descriptive ones? e.g. plaintext, ciphertext, public_key, secret_key etc

Reply:
I think this small and rough utility should **not** be used as a reference in libsodium C usage, especially since I'm not a native C "speaker" and therefore I'd like to avoid any not strictly needed effort on it's implementation.